SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -460,13 +457,52 @@ void ASpatialDebugger::DrawTag(UCanvas* Canvas, const FVector2D& ScreenLocation,
 
 	const SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(EntityId);
 
-	static const float BaseHorizontalOffset(16.0f);
-
 	if (!FApp::CanEverRender()) // DrawIcon can attempt to use the underlying texture resource even when using nullrhi
 	{
 		return;
 	}
 
+	static const float BaseHorizontalOffset(16.0f);
+	static const float NumberScale = 0.75f;
+	static const float TextScale = 0.5f;
+	const float AuthWidth = NumberScale * GetNumberOfDigitsIn(DebuggingInfo->AuthoritativeVirtualWorkerId);
+	const float AuthIntentWidth = NumberScale * GetNumberOfDigitsIn(DebuggingInfo->IntentVirtualWorkerId);
+	const float EntityIdWidth = NumberScale * GetNumberOfDigitsIn(EntityId);
+
+	int32 HorizontalOffset = 0;
+	if (bCentre)
+	{
+		// If tag should be centered, calculate the total width of the icons and text to be rendered
+		float TagWidth = 0;
+		if (bShowLock)
+		{
+			TagWidth += BaseHorizontalOffset;
+		}
+		if (bShowAuth)
+		{
+			TagWidth += BaseHorizontalOffset;
```

Comment:
Why do we double increment here? Shouldn't AuthWidth just be +=1.f ?

Reply:
Oh it's used below

Reply:
Still a little confused - is this additional spacing for something?

Reply:
The Authority and the AuthorityIntent have an icon and text so they need two widths. The icons are a fixed size and the text depends on the number of digits in the ID.

Reply:
I will add some comments to make it clearer.

Reply:
I've added comments and updated variable names to make the tag width calculation clearer.